Fix preview of some gif URLs (#11669)
matrix-org/synapse: docs/development/url_previews.md

      2. Generates an Open Graph response based on image properties.
   5. If the media is HTML:
      1. Decodes the HTML via the stored file.
      2. Generates an Open Graph response from the HTML.
      3. If an image exists in the Open Graph response:
         1. Downloads the URL and stores it into a file via the media storage
            provider and saves the local media metadata.
         2. Generates thumbnails.
         3. Updates the Open Graph response based on image properties.

>    6. If the media is JSON and an oEmbed URL was found:

Reviewer: This part is still valid if the oEmbed is downloaded via direct configuration instead of auto-discovery.

Author: Oh I see, this code is still not completely clear to me, but I'll try a fix

Reviewer: Yeah, it is a bit confusing (which is why I tried to document it!) It seems that I failed to update it when I added the auto-discovery though.

I wonder if a flow-chart would be clearer...

Please feel free to reach out in #synapse-dev:matrix.org if you have questions! I'm more than happy to help!

      2. Generates an Open Graph response based on image properties.
   5. If the media is HTML:
      1. Decodes the HTML via the stored file.

>       2. If a JSON oEmbed URL was found in the HTML:
>          1. Convert the oEmbed response to an Open Graph response.
>          2. If a thumbnail or image is in the oEmbed response:
>             1. Downloads the URL and stores it into a file via the media storage
>                provider and saves the local media metadata.
>             2. Generates thumbnails.
>             3. Updates the Open Graph response based on image properties.
>          3. If the oEmbed type is video but no video is provided, abort oEmbed
>             parsing and fall back to Open Graph
>       3. Generates an Open Graph response from the HTML.
>       4. If an image exists in the Open Graph response:

Reviewer: I don't think this is accurate anymore:

```suggestion
      2. Generates an Open Graph response from the HTML.
      3. If a JSON oEmbed URL was found in the HTML via autodiscovery:
         1. Downloads the URL and stores it into a file via the media storage provider
            and saves the local media metadata.
         2. Convert the oEmbed response to an Open Graph response.
         3. Override any Open Graph data from the HTML with data from oEmbed.
      4. If an image exists in the Open Graph response:
```

         1. Downloads the URL and stores it into a file via the media storage
            provider and saves the local media metadata.
         2. Generates thumbnails.
         3. Updates the Open Graph response based on image properties.
   6. If the media is JSON and an oEmbed URL was found:

>       1. Parse it as described in 3.5.2

Reviewer: I think this should be left unmodified, even if it is a bit repetitive.
